src/Mvc/test/WebSites/RoutingWebSite/StartupForGroups.cs

```diff
@@ -29,7 +29,11 @@ public virtual void Configure(IApplicationBuilder app)
             pagesGroup.MapRazorPages();
 
             var controllerGroup = endpoints.MapGroup("/controllers/{org}");
-            controllerGroup.MapControllers();
+            controllerGroup.MapControllers().AddEndpointFilter((context, next) =>
+            {
+                context.HttpContext.Items[nameof(IEndpointFilter)] = true;
```

Comment:
Maybe also verify the method info. That's one of the things that changes would be good to assert.